source/geometries/Next100Vessel.cc

```diff
@@ -120,112 +124,166 @@ namespace nexus {
-							  0. * cm,  endcap_out_rad,   //radius
-							  0. * deg, 360. * deg,       // phi
-							  0. * deg, endcap_theta_);   // theta
-
-    G4Sphere* vessel_gas_tracking_endcap_solid = new G4Sphere("VESSEL_GAS_TRACKING_ENDCAP",
-							      0. * cm,  endcap_in_rad_,   //radius
-							      0. * deg, 360. * deg,       // phi
-							      0. * deg, endcap_theta_);   // theta
+    G4double endcap_out_rad = endcap_in_rad_ + vessel_thickness_;
 
-    G4Sphere* vessel_energy_endcap_solid = new G4Sphere("VESSEL_ENERGY_ENDCAP",
-							0. * cm,  endcap_out_rad,                   //radius
-							0. * deg, 360. * deg,                       // phi
-							180. * deg - endcap_theta_, endcap_theta_); // theta
-
-    G4Sphere* vessel_gas_energy_endcap_solid = new G4Sphere("VESSEL_GAS_ENERGY_ENDCAP",
-							    0. * cm,  endcap_in_rad_,                   //radius
-							    0. * deg, 360. * deg,                       // phi
-							    180. * deg - endcap_theta_, endcap_theta_); // theta
+    G4Sphere* vessel_endcap_solid = new G4Sphere("VESSEL_ENDCAP",
+							  0. * cm,  endcap_out_rad,
+							  0. * deg, 360. * deg,
+							  0. * deg, endcap_theta_);
 
+    G4Sphere* vessel_gas_endcap_solid = new G4Sphere("VESSEL_GAS_ENDCAP",
+							      0. * cm,  endcap_in_rad_,
+							      0. * deg, 360. * deg,
+							      0. * deg, endcap_theta_);
 
     // Flange solid
-    G4Tubs* vessel_flange_solid = new G4Tubs("VESSEL_TRACKING_FLANGE", 0., flange_out_rad_,
-						      flange_length_/2., 0.*deg, 360.*deg);
-
-    // Nozzle solids
-    G4Tubs* large_nozzle_solid = new G4Tubs("LARGE_NOZZLE", 0.*cm, nozzle_ext_diam_/2.,
-					    large_nozzle_length_/2., 0.*deg, 360.*deg);
-
-    G4Tubs* small_nozzle_solid = new G4Tubs("SMALL_NOZZLE", 0.*cm, nozzle_ext_diam_/2.,
-					    small_nozzle_length_/2., 0.*deg, 360.*deg);
-
-    G4Tubs* large_nozzle_gas_solid = new G4Tubs("LARGE_NOZZLE_GAS", 0.*cm, (nozzle_ext_diam_/2. - vessel_thickness_),
-						large_nozzle_length_/2., 0.*deg, 360.*deg);
-
-    G4Tubs* small_nozzle_gas_solid = new G4Tubs("SMALL_NOZZLE_GAS", 0.*cm, (nozzle_ext_diam_/2. - vessel_thickness_),
-						small_nozzle_length_/2., 0.*deg, 360.*deg);
+    G4double flange_out_rad   = 74.0 * cm;
+    G4double flange_tp_length = 41.5 * mm + 41.5 * mm; // 41.5 mm from the endcap
+    G4double flange_ep_length = 77.5 * mm + 41.5 * mm;
+    G4double flange_ep_z_pos  =   body_length_/2. - flange_ep_length/2. - endcap_in_body_;
+    G4double flange_tp_z_pos  = -(body_length_/2. - flange_tp_length/2. - endcap_in_body_);
+
+    G4Tubs* vessel_tp_flange_solid = new G4Tubs("VESSEL_TRACKING_FLANGE", vessel_in_rad_, flange_out_rad,
+                                                flange_tp_length/2., 0.*deg, 360.*deg);
+    G4Tubs* vessel_ep_flange_solid = new G4Tubs("VESSEL_ENERGY_FLANGE"  , vessel_in_rad_, flange_out_rad,
+                                                flange_ep_length/2., 0.*deg, 360.*deg);
+
+    // Calibration ports (approximate values)
```

Comment:
You can get the exact values from '000-90.pdf' (4 and 5, in LL)

Reply:
I have updated the values following this 2D drawing, except the heights that can not be infered from it, but can be easily measured in the 3D drawing.